### PR TITLE
Add pre-commit for golang

### PR DESCRIPTION
**Pull request by fedeizzo**
What does this PR do?
-----------------------

- Move `go.mod` and `go.sum` files to the root of the project.
- Add [pre-commit](https://pre-commit.com/) definition for golang:
   - `govet`
   - `golangci-lint`
   - `goimports`
   - `gofmt`
- Update `CONTRIBUTING.md` guidelines

Testing
-------
```bash
nix flake check
```
or
```bash
pre-commit install
pre-commit run --all-files
```

### CODE REVIEW

**Review by luigibarbato — DISMISSED** [review]
SBAM 💣

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -31,6 +31,15 @@ just web/watch
 
 The server should be listening on http://127.0.0.1:4000, while the web application should be available on http://127.0.0.1:8080.
 
+## Pre-commit
+Each commit is run against a list of checks defined using [pre-commit](https://pre-commit.com/). Before contribuiting to this project, be sure to install them.
+
+### Nix
+If you use nix everything you are ready to go, [git-hooks.nix](https://github.com/cachix/git-hooks.nix/tree/master) and [flake.parts](https://flake.parts/) take care of everything.
```
Comment (MatteoJoliveau) [inline CONTRIBUTING.md:38]:
```suggestion
If you use Nix for everything you are ready to go, [git-hooks.nix](https://github.com/cachix/git-hooks.nix/tree/master) and [flake.parts](https://flake.parts/) take care of everything.
```

File: flake.nix [hunk 2]
```diff
@@ -5,33 +5,99 @@
...
+                };
+              };
+            };
+          };
+
           formatter = pkgs.nixpkgs-fmt;
 
           devShells.default = pkgs.mkShell {
             PLAYWRIGHT_BROWSERS_PATH = pkgs.playwright-driver.browsers;
+            buildInputs = config.pre-commit.settings.enabledPackages;
```
Comment (MatteoJoliveau) [inline flake.nix:85]: Doesn't this override the `devShells.default.packages` attribute?

**Review by MatteoJoliveau — CHANGES_REQUESTED** [review]

File: flake.nix (hunk 2 above)
Comment (fedeizzo) [reply flake.nix:85]: You are right, wrong place.

I am moving `enabledPackages` under `packages` and everything works as expected..

```sh
$ which golangci-lint
/nix/store/5pjiagw5zn3xfws2wdnaw094p88n6jaa-golangci-lint-1.57.2/bin/golangci-lint
$ nix flake check && echo $?
0
```

**Review by MatteoJoliveau — APPROVED** [review]
LGTM 🚀

**Pull request merged**